locust/main.py

```diff
@@ -417,17 +417,15 @@ def shutdown():
     # install SIGTERM handler
     def sig_term_handler():
         logger.info("Got SIGTERM signal")
-        shutdown()
```

Comment:
Hmm, won't removing this call stop `shutdown()` from being called when the process receives a `TERM` signal?

Reply:
> I believe I've fixed this in [cb56d64](https://github.com/locustio/locust/commit/cb56d64f158bd8df2d833b9f252836d19dd39fbe) (I had to get all the tests to pass when migrating CI from Travis to Github Actions).
> 
> Could you confirm if that fixes your issue?

I can confirm that the issue was fixed proper with your change. It's almost the same code. I have updated this PR in order to be able to see the final changes. But I think that it could be closed if you don't want to merge the change of the `finally`. Let me know

Reply:
the finally clause will warranty the execution of the shutdown method

Reply:
I could be wrong, but I don't think that the `main_greenlet.join()` will return unless shutdown() is called.

Reply:
You aren't wrong... but I think that in order to warranty the usage of the html report option the finally need to be used or like you write below move the code to the shutdown method will work too